src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -1521,12 +1521,16 @@ public virtual void Clear()
             // have invalid inputs, so these executions are meaningless and may
             // cause invalid GC. See comments in MAGN-7229.
             foreach (NodeModel node in Nodes)
+            {
                 node.RaisesModificationEvents = false;
+                // Dispose here so that all nodes stop listening to disconnect events before
+                // the connectors are deleted. Otherwise remaining undisposed nodes will react
+                // to delete events when an input connector is deleted.
+                node.Dispose();
+            }
 
             foreach (NodeModel el in Nodes)
```

Comment:
Why can't both these `for `loops be combined?

Reply:
Understood. This brings me to my next question - shouldn't the nodes be unsubscribed from those events before they are deleted? There could be a memory leak if they are not unsubscribed?